core/domain/question_services.py

```diff
@@ -217,3 +228,319 @@ def update_question(
+
+    question_models.QuestionRightsModel(
+        id=question_rights.id,
+        manager_ids=question_rights.manager_ids
+    ).commit(committer_id, 'Created new question rights', commit_cmds)
+
+
+def get_question_rights(question_id, strict=True):
+    """Retrieves the rights object for the given question.
+
+    Args:
+        question_id: str. ID of the question.
+        strict: bool. Whether to fail noisily if no question rights with a
+            given id exists in the datastore.
+
+    Returns:
+        QuestionRights. The rights object associated with the given question.
+
+    Raises:
+        EntityNotFoundError. The question with ID question_id was not
+            found in the datastore.
+    """
+
+    model = question_models.QuestionRightsModel.get(question_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_question_rights_from_model(model)
+
+
+def check_can_edit_question(user_id, question_id):
+    """Checks if the user can edit the given question or not
+
+    Args:
+        user_id: str. The user ID of the user.
+        question_id: str. The question ID of the question.
+
+    Returns:
+        bool. Represents if the user can edit the question or not.
+    """
+    question_summary = get_question_summary_by_question_id(
+        question_id, strict=False)
+
+    if question_summary.status == feconf.QUESTION_STATUS_PENDING:
+        return False
+
+    if (user_services.is_admin(user_id) or
+            user_services.is_topic_manager(user_id)):
+        if question_summary.status == feconf.QUESTION_STATUS_APPROVED:
+            return True
+        return False
+
+    if (question_summary.status == feconf.ACTIVITY_STATUS_PRIVATE or
+            question_summary.status == feconf.QUESTION_STATUS_REJECTED):
+        return True
+
+    return False
+
+
+def publish_question(question_id, committer_id):
+    """Marks the given question as published.
+
+    Args:
+        question_id: str. The id of the given question.
+        committer_id: str. ID of the committer.
+
+    Raises:
+        Exception. The given question does not exist.
+        Exception. The question is already published.
+        Exception. The user does not have enough rights to publish the question.
+    """
+    question_summary = get_question_summary_by_question_id(
+        question_id, strict=False)
+    if question_summary is None:
+        raise Exception('The given question does not exist')
+    user = user_services.UserActionsInfo(committer_id)
+    if role_services.ACTION_CHANGE_QUESTION_STATUS not in user.actions:
+        raise Exception(
+            'The user does not have enough rights to publish the question.')
+
+    if question_summary.status == feconf.QUESTION_STATUS_APPROVED:
+        raise Exception('The question is already published.')
+    question_summary.status = feconf.QUESTION_STATUS_APPROVED
+    question_summary.put()
+
+
+def unpublish_question(question_id, committer_id):
```

Comment:
This  poses a question. After a  question is unpublished, can the initial creator of the question edit it, or is it that once it has been approved, the creator has lost all privileges?

Reply:
Once the question is published, the initial creator will lose all the rights. That means once the question is approved/published, the admin and topic managers become the maintainer of the Question

Reply:
Ok, I think that's ok.